### PR TITLE
posix: implement putpmsg() 

### LINKED ISSUE #66979: posix: implement `putpmsg()`
This is part of the See [POSIX Roadmap for LTSv3](https://github.com/zephyrproject-rtos/zephyr/issues/51211) (RFC #51211).

`putpmsg()` is required as part of `_XOPEN_STREAMS` Option Group.

For more information, please refer to https://pubs.opengroup.org/onlinepubs/9699919799/functions/putpmsg.html

### PR DESCRIPTION
**Pull request by AbhinavMir**
Fixes https://github.com/zephyrproject-rtos/zephyr/issues/66979

### CODE REVIEW

**Comment by AbhinavMir** [comment]
@ycsin I messed up my previous branch a little too much, so just did a fresh branch commit, this should be good compliance wise, and updated to the latest upstream.

File: include/zephyr/posix/stropts.h [hunk 1]
```diff
@@ -19,6 +19,9 @@ struct strbuf {
 
 int putmsg(int fildes, const struct strbuf *ctlptr, const struct strbuf *dataptr, int flags);
 
+int putpmsg(int fildes, const struct strbuf *ctlptr,
```
Comment (ycsin) [inline include/zephyr/posix/stropts.h:22]: format is off

File: lib/posix/options/stropts.c [hunk 2]
```diff
@@ -18,3 +18,16 @@ int putmsg(int fildes, const struct strbuf *ctlptr, const struct strbuf *dataptr
 	errno = ENOSYS;
 	return -1;
 }
+
+int putpmsg(int fildes, const struct strbuf *ctlptr,
+const struct strbuf *dataptr, int band, int flags)
```
Comment (ycsin) [inline lib/posix/options/stropts.c:23]: format is off

File: tests/posix/headers/src/stropts_h.c [hunk 3]
```diff
@@ -26,3 +26,10 @@ ZTEST(posix_headers, test_stropts_h)
 	zassert_not_equal(RS_HIPRI, ~RS_HIPRI);
 	#endif
 }
+
+ZTEST(posix_headers, test_stropts_putpmsg)
+{
+	#ifdef CONFIG_POSIX_API
```
Comment (ycsin) [inline tests/posix/headers/src/stropts_h.c:32]: we usually do not indent preprocessor macros

**Review by ycsin — CHANGES_REQUESTED** [review]

File: lib/posix/options/stropts.c (hunk 2 above)
Comment (AbhinavMir) [reply lib/posix/options/stropts.c:23]: Do I need to break a long line? I did the warning from my IDE that this line is too long.

File: tests/posix/headers/src/stropts_h.c (hunk 3 above)
Comment (AbhinavMir) [reply tests/posix/headers/src/stropts_h.c:32]: Should there be any form of indentation? Or would it be
```
ZTEST(posix_headers, test_stropts_putpmsg)
{
#ifdef CONFIG_POSIX_API
```

Comment (ycsin) [reply tests/posix/headers/src/stropts_h.c:32]: Yeah should be something like that

**Comment by ycsin** [comment]
You can change the PR status to 'draft' if it is still a WIP

**Pull request closed without merging**